src/main/java/at/hannibal2/skyhanni/features/garden/pests/PestFinder.kt

```diff
@@ -139,54 +211,96 @@ class PestFinder {
     fun onScoreboardChange(event: ScoreboardChangeEvent) {
         if (!GardenAPI.inGarden()) return
 
-        var newPests = 0
         for (line in event.newList) {
             pestsInScoreboardPattern.matchMatcher(line) {
-                newPests = group("pests").formatNumber().toInt()
+                val newPests = group("pests").formatNumber().toInt()
+                if (newPests != PestAPI.scoreboardPests) {
+                    removePests(PestAPI.scoreboardPests - newPests)
+                    PestAPI.scoreboardPests = newPests
+                    fixPests()
+                    update()
+                }
+            }
+            noPestsInScoreboardPattern.matchMatcher(line) {
+                if (PestAPI.scoreboardPests != 0) {
+                    PestAPI.scoreboardPests = 0
+                    GardenPlotAPI.plots.forEach {
+                        it.pests = 0
+                        it.isPestCountInaccurate = false
+                    }
```

Comment:
put those lines into a function, and use it two times. put the update call at the end in the function. call it something like resetAllPests idk